EIPS/eip-2200.md

```diff
@@ -68,29 +98,221 @@ the following logic:
+
+* For *absolute gas used* (that is, actual *gas used* minus *refund*),
+  this EIP is equivalent to EIP-1087 for all cases.
+* For one particular case, where a storage slot is changed, reset to
+  its original value, and then changed again, EIP-1283 would move more
+  gases to refund counter compared with EIP-1087.
+
+Examine examples provided in EIP-1087's Motivation:
+
+* If a contract with empty storage sets slot 0 to 1, then back to 0,
+  it will be charged `20000 + 200 - 19800 = 400` gas.
+* A contract with empty storage that increments slot 0 5 times will be
+  charged `20000 + 5 * 200 = 21000` gas.
+* A balance transfer from account A to account B followed by a
+  transfer from B to C, with all accounts having nonzero starting and
+  ending balances, it will cost `5000 * 3 + 200 - 4800 = 10400` gas.
+
+In order to keep in place the implicit reentrancy protection of
+existing contracts, transactions should not be allowed to modify state
+if the remaining gas is lower then the gas stipend given to
+"transfer"/"send" in Solidity. These are other proposed remediations
+and objections to implementing them:
+
+* Drop EIP-1283 and abstain from modifying SSTORE cost
+  * EIP-1283 is an important update 
+  * It was accepted and implemented on test networks and in clients.
+* Add a new call context that permits LOG opcodes but not changes to state.
+  * Adds another call type beyond existing regular/staticcall
+* Raise the cost of SSTORE to dirty slots to >=2300 gas
+  * Makes net gas metering much less useful.
+* Reduce the gas stipend
+  * Makes the stipend almost useless.
+* Increase the cost of writes to dirty slots back to 5000 gas, but add
+  4800 gas to the refund counter
+  * Still doesn’t make the invariant explicit.
+  * Requires callers to supply more gas, just to have it refunded
+* Add contract metadata specifying per-contract EVM version, and only
+  apply SSTORE changes to contracts deployed with the new version.
 
 ## Backwards Compatibility
 
-This EIP has the same backward compatibility property of EIP-1283 and EIP-1706.
+This EIP requires a hard fork to implement. No gas cost increase is
+anticipated, and many contracts will see gas reduction.
+
+Performing SSTORE has never been possible with less than 5000 gas, so
+it does not introduce incompatibility to the Ethereum mainnet. Gas
+estimation should account for this requirement.
 
 ## Test Cases
 
-To be added.
+Below we provide 17 test cases. 15 of them covering consecutive two
+`SSTORE` operations are based on work [by
+@chfast](https://github.com/ethereum/tests/issues/483). Two additional
+case with three `SSTORE` operations is used to test the case when a
+slot is reset and then set again.
+
+| Code                               | Used Gas | Refund | Original | 1st | 2nd | 3rd |
+|------------------------------------|----------|--------|----------|-----|-----|-----|
+| `0x60006000556000600055`           | 412      | 0      | 0        | 0   | 0   |     |
```

Comment:
These are old test vectors from EIP-1283 that assume SLOAD to cost 200.
Here are the new test vectors with SLOAD cost 800 https://github.com/sorpaas/EIPs/issues/1#issuecomment-517305533

Reply:
Updated!